src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Utf8JsonWriterTests.WriteRaw.cs

```diff
@@ -0,0 +1,259 @@
+            }
+        }
+
+        /// <summary>
+        /// This test is constrained to run on Windows and MacOSX because it causes
+        /// problems on Linux due to the way deferred memory allocation works. On Linux, the allocation can
+        /// succeed even if there is not enough memory but then the test may get killed by the OOM killer at the
+        /// time the memory is accessed which triggers the full memory allocation.
+        /// Also see <see cref="Utf8JsonWriterTests.WriteLargeJsonToStreamWithoutFlushing"/>
+        /// </summary>
+        [PlatformSpecific(TestPlatforms.Windows | TestPlatforms.OSX)]
+        [ConditionalFact(nameof(IsX64))]
+        [OuterLoop]
+        public void WriteLargeRawJsonToStreamWithoutFlushing()
+        {
+            var largeArray = new char[150_000_000];
+            largeArray.AsSpan().Fill('a');
+
+            // Text size chosen so that after several doublings of the underlying buffer we reach ~2 GB (but don't go over)
+            JsonEncodedText text1 = JsonEncodedText.Encode(largeArray.AsSpan(0, 7_500));
+            JsonEncodedText text2 = JsonEncodedText.Encode(largeArray.AsSpan(0, 5_000));
+            JsonEncodedText text3 = JsonEncodedText.Encode(largeArray.AsSpan(0, 150_000_000));
+
+            using (var output = new MemoryStream())
+            using (var writer = new Utf8JsonWriter(output))
+            {
+                writer.WriteStartArray();
+                writer.WriteRawValue(text1.EncodedUtf8Bytes);
+                Assert.Equal(7_503, writer.BytesPending);
+
+                for (int i = 0; i < 30_000; i++)
+                {
+                    writer.WriteRawValue(text2.EncodedUtf8Bytes);
+                }
+                Assert.Equal(150_097_503, writer.BytesPending);
+
+                for (int i = 0; i < 13; i++)
+                {
+                    writer.WriteRawValue(text3.EncodedUtf8Bytes);
+                }
+                Assert.Equal(2_100_097_542, writer.BytesPending);
+
+                // Next write forces a grow beyond max array length
+
+                Assert.Throws<OutOfMemoryException>(() => writer.WriteRawValue(text3.EncodedUtf8Bytes));
+
+                Assert.Equal(2_100_097_542, writer.BytesPending);
+
+                var text4 = JsonEncodedText.Encode(largeArray.AsSpan(0, 1));
+                for (int i = 0; i < 10_000_000; i++)
+                {
+                    writer.WriteRawValue(text4.EncodedUtf8Bytes);
+                }
+
+                Assert.Equal(2_100_097_542 + (4 * 10_000_000), writer.BytesPending);
+            }
+        }
+
+        [Fact]
+        public static void WriteRawNullOrEmptyTokenInvalid()
+        {
+            using MemoryStream ms = new();
+            using Utf8JsonWriter writer = new(ms);
+            Assert.Throws<ArgumentNullException>(() => writer.WriteRawValue(json: default(string)));
+            Assert.Throws<ArgumentException>(() => writer.WriteRawValue(json: ""));
```

Comment:
What about just whitespace but non-empty string: `"    "`

Reply:
Should fail on reader validation when `skipInputValidation` is false. Added a test to `WriteRawInvalidJson`.